src/QMCWaveFunctions/LCAO/MultiQuinticSpline1D.h

```diff
@@ -137,6 +152,68 @@ class MultiQuinticSpline1D
     }
   }
 
+  /**
+   * @brief evaluate MultiQuinticSpline1D for multiple electrons and multiple pbc images
+   * 
+   * @param [in] r electron distances [Nelec, Npbc]
+   * @param [out] u value of all splines at all electron distances [Nelec, Npbc, Nsplines]
+   * @param Rmax spline will evaluate to zero for any distance greater than or equal to Rmax
+  */
+  inline void batched_evaluate(const OffloadArray2D& r, OffloadArray3D& u, T Rmax) const
+  {
+    const size_t nElec = r.size(0);
+    const size_t Nxyz  = r.size(1); // number of PBC images
+    assert(nElec == u.size(0));
+    assert(Nxyz == u.size(1));
+    const size_t nRnl = u.size(2);    // number of splines
+    const size_t nR   = nElec * Nxyz; // total number of positions to evaluate
+
+    double one_over_log_delta = myGrid.OneOverLogDelta;
+    T lower_bound             = myGrid.lower_bound;
+    T log_delta               = myGrid.LogDelta;
+
+    auto* r_ptr = r.data();
+    auto* u_ptr = u.data();
+
+    auto* coeff_ptr       = coeffs->data();
+    auto* first_deriv_ptr = first_deriv.data();
+    const size_t nCols    = coeffs->cols();
+    const size_t coefsize = coeffs->size();
+
+
+    PRAGMA_OFFLOAD("omp target teams distribute parallel for \
+                    map(always, to:coeff_ptr[:coefsize], first_deriv_ptr[:num_splines_]) \
```

Comment:
Could you add a FIXME saying the "always" should be removed after making the transfer only once ahead of time?

Reply:
fixed